src/client.rs

```diff
@@ -50,6 +50,18 @@ impl Client {
         }
     }
 
+    /// Close the client.
+    /// Closing the client is not required but doing so will ensure that no more messages can be
+    /// produced. The receiver of the messages will be able to consume any in-flight messages and
+    /// then will observe the end of the stream.
+    ///
+    /// If the client is never closed and never dropped the reveiver of the messages will never observe the end of
+    /// the stream.
+    pub fn close(&self) {
```

Comment:
I don't think we want to expose this method to downstream users. Ideally, the `exit` notification alone should dictate when communications between the server and client shut down. I understand this method is only really used in the codegen output of `tower-lsp-macros`. Perhaps we could consider changing the visibility to `pub(crate)` instead to avoid exposing this in the public API?